### PR TITLE
[Single File] Add single file support for Flux Transformer

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
Add single file support for the Flux Transformer model. 

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/loaders/single_file_utils.py [hunk 1]
```diff
@@ -1859,3 +1867,195 @@ def convert_animatediff_checkpoint_to_diffusers(checkpoint, **kwargs):
...
     return converted_state_dict
+
+
+def convert_flux_transformer_checkpoint_to_diffusers(checkpoint, **kwargs):
+    converted_state_dict = {}
+
+    num_layers = list(set(int(k.split(".", 2)[1]) for k in checkpoint if "double_blocks." in k))[-1] + 1  # noqa: C401
+    num_single_layers = list(set(int(k.split(".", 2)[1]) for k in checkpoint if "single_blocks." in k))[-1] + 1  # noqa: C401
+    mlp_ratio = 4.0
+    inner_dim = 3072
```
Comment (DN6) [inline src/diffusers/loaders/single_file_utils.py:1878]: Should be okay to hardcode these no? They are the same across both models. We can grab `inner_dim` from the checkpoint, but not sure about `mlp_ratio`?

Comment (DN6) [reply src/diffusers/loaders/single_file_utils.py:1878]: cc: @yiyixuxu

Comment (sayakpaul) [reply src/diffusers/loaders/single_file_utils.py:1878]: I think okay to hard code. I would just define them as proper constants at the top of the file.

File: src/diffusers/loaders/single_file_utils.py [hunk 2]
```diff
@@ -1859,3 +1867,195 @@ def convert_animatediff_checkpoint_to_diffusers(checkpoint, **kwargs):
...
+
+    # in SD3 original implementation of AdaLayerNormContinuous, it split linear projection output into shift, scale;
```
Comment (sayakpaul) [inline src/diffusers/loaders/single_file_utils.py:1880]: Can leave outside of the function no since SD3 also uses it?

**Review by sayakpaul — APPROVED** [review]
LGTM!

Two things:

* Let's add it to the list of models that support single file loading. 
* Let's see if FP8 support works? If so, let's attach a code snippet?

File: src/diffusers/loaders/single_file_utils.py (hunk 1 above)
Comment (DN6) [reply src/diffusers/loaders/single_file_utils.py:1878]: But if they're only applied within the scope of this function, they don't need to exist as constants that can be accessed globally?

**Comment by sayakpaul** [comment]
@DN6 let's merge this after the TODOs?

**Pull request merged**